loki-client/src/main/java/com/github/loki4j/client/pipeline/PipelineConfig.java

```diff
@@ -187,6 +197,17 @@ public static class Builder {
         private long sendQueueMaxBytes = batchMaxBytes * 10;
         private int maxRetries = 2;
         private long retryTimeoutMs = 60 * 1000;
+        private Supplier<Long> jitter = new Jitter();
+        private BiFunction<Integer, Long, Boolean> sleep = (attempt, timeout) -> {
```

Comment:
I see your point, but I don't think we would need this kind of flexibility in the nearest future. Thus, I'd avoid adding the things we don't really need to configure to the configuration.

Reply:
I think allowing to override sleep is key, otherwise your test will be unnecessarily slow. The alternative is to have some sleeps in the tests which is a bad test practice and honestly makes the test unreliable and slow.